db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -1331,6 +1334,11 @@ Status DBImpl::ReFitLevel(ColumnFamilyData* cfd, int level, int target_level) {
     }
   }
   if (to_level != level) {
+    // This might also be checked previously in some cases, but still double
+    // check here to prevent corruption..
+    if (vstorage->NumLevelFiles(to_level) > 0) {
```

Comment:
Do we need to check emptiness of all levels in `[to_level, level)`?

Reply:
Good point. I think it is a good idea. Let me do it.